homeassistant/components/bmw_connected_drive/binary_sensor.py

```diff
@@ -16,89 +19,101 @@
+        name="Parking lights",
+        device_class="light",
+        icon="mdi:car-parking-lights",
+    ),
+    BinarySensorEntityDescription(
+        key="condition_based_services",
+        name="Condition based services",
+        device_class=DEVICE_CLASS_PROBLEM,
+        icon="mdi:wrench",
+    ),
+    BinarySensorEntityDescription(
+        key="check_control_messages",
+        name="Control messages",
+        device_class=DEVICE_CLASS_PROBLEM,
+        icon="mdi:car-tire-alert",
+    ),
+    # electric
+    BinarySensorEntityDescription(
+        key="charging_status",
+        name="Charging status",
+        device_class="power",
+        icon="mdi:ev-station",
+    ),
+    BinarySensorEntityDescription(
+        key="connection_status",
+        name="Connection status",
+        device_class=DEVICE_CLASS_PLUG,
+        icon="mdi:car-electric",
+    ),
+)
 
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the BMW ConnectedDrive binary sensors from config entry."""
     account = hass.data[BMW_DOMAIN][DATA_ENTRIES][config_entry.entry_id][CONF_ACCOUNT]
-    entities = []
-
-    for vehicle in account.account.vehicles:
-        if vehicle.has_hv_battery:
-            _LOGGER.debug("BMW with a high voltage battery")
-            for key, value in sorted(SENSOR_TYPES_ELEC.items()):
-                if key in vehicle.available_attributes:
-                    device = BMWConnectedDriveSensor(
-                        account, vehicle, key, value[0], value[1], value[2]
-                    )
-                    entities.append(device)
-        elif vehicle.has_internal_combustion_engine:
-            _LOGGER.debug("BMW with an internal combustion engine")
-            for key, value in sorted(SENSOR_TYPES.items()):
-                if key in vehicle.available_attributes:
-                    device = BMWConnectedDriveSensor(
-                        account, vehicle, key, value[0], value[1], value[2]
-                    )
-                    entities.append(device)
+
+    entities = [
+        BMWConnectedDriveSensor(account, vehicle, description)
+        for vehicle in account.account.vehicles
+        for description in SENSOR_TYPES
+        if description.key in vehicle.available_attributes
+    ]
     async_add_entities(entities, True)
 
 
 class BMWConnectedDriveSensor(BMWConnectedDriveBaseEntity, BinarySensorEntity):
     """Representation of a BMW vehicle binary sensor."""
 
-    def __init__(
-        self, account, vehicle, attribute: str, sensor_name, device_class, icon
-    ):
+    def __init__(self, account, vehicle, description: BinarySensorEntityDescription):
         """Initialize sensor."""
         super().__init__(account, vehicle)
+        self.entity_description = description
 
-        self._attribute = attribute
-        self._attr_name = f"{vehicle.name} {attribute}"
-        self._attr_unique_id = f"{vehicle.vin}-{attribute}"
-        self._sensor_name = sensor_name
```

Comment:
`self._sensor_name` wasn't assigned to anything.
It could make sense to do `self._attr_name = f"{vehicle.name} {description.name}"` now.